### PR TITLE
Fixed Tall Grass placement

### PR DESCRIPTION
**Pull request by SamJBarney**
* Fixed tall grass trying to be placed in Ice Plains, Cold Beach, and Ice Mountains biomes.
* Fixes https://github.com/cuberite/cuberite/issues/2265

### CODE REVIEW

File: src/Generating/FinishGen.cpp [hunk 1]
```diff
@@ -337,7 +340,7 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
 			{
 				a_ChunkDesc.SetBlockTypeMeta(x, y, z, E_BLOCK_TALL_GRASS, 2);
 			}
-			else
+			else if ((biome != biIcePlains) && (biome != biColdBeach) && (biome != biIceMountains) && (biome != biIcePlainsSpikes))
```
Comment (NiLSPACE) [inline src/Generating/FinishGen.cpp]: I believe we have a `IsBiomeVeryCold` function for this. It's defined in BiomeDef I believe.

File: src/Generating/FinishGen.cpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #include "../Simulator/FireSimulator.h"
 #include "../World.h"
 #include "../IniFile.h"
+#include "../BiomeDef.h"
```
Comment (NiLSPACE) [inline src/Generating/FinishGen.cpp]: It *could* be possible that this is also included in the globals. Did you try to compile it without this include?

Comment (SamJBarney) [inline src/Generating/FinishGen.cpp]: Just tested it. Works without it.

Comment (NiLSPACE) [inline src/Generating/FinishGen.cpp]: Then only now a squash can be done, and it can be merged ;)

**Comment by NiLSPACE** [comment]
Great that this bug is getting fixed :) It was getting on my nerves, but since I usually avoid snow biomes I didn't see it allot, and thus I forgot about it.

**Comment by SamJBarney** [comment]
Is this one ok to merge?

File: src/Generating/FinishGen.cpp [hunk 3]
```diff
@@ -322,6 +322,7 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
 			if (
 				(a_ChunkDesc.GetBlockType(x, y, z) != E_BLOCK_AIR) ||
 				((a_ChunkDesc.GetBlockType(x, y - 1, z) != E_BLOCK_GRASS) && (a_ChunkDesc.GetBlockType(x, y - 1, z) != E_BLOCK_DIRT))
+
```
Comment (tigerw) [inline src/Generating/FinishGen.cpp]: random carriage return is random

**Comment by tigerw** [comment]
It presumably works?

**Comment by SphinxC0re** [comment]
What do you mean? ... oh :laughing: yeah, squash and rebase onto current master

**Comment by tigerw** [comment]
Precisely my dear fellow.

**Comment by SamJBarney** [comment]
Squashed it. Sorry, I guess I needed to be more clear with what I meant to ask. I meant to ask if there were any other changes needed before I squashed it.

**Comment by SphinxC0re** [comment]
Merge if compiles :smile:

**Comment by tigerw** [comment]
>  I meant to ask if there were any other changes needed before I squashed it.

If you do want to fix something, I added a comment on a wild newline appeared above.

**Comment by SphinxC0re** [comment]
@tigerw for real? It's a newline.

**Comment by SamJBarney** [comment]
A Wild Newline appears!
@tigerw used Scary Face!
The Wild Newline ran away!

----
Sorry. Forgot about that one overnight, tigerw.

**Comment by SphinxC0re** [comment]
Oh, @tigerw will push it to the limit! *Baaaaaaaaaaaam!*

**Comment by tigerw** [comment]
[Shakes head](http://www.reactiongifs.us/wp-content/uploads/2013/08/shaking_head_breaking_bad.gif)

**Pull request merged**